tritond/mock.go

```diff
@@ -37,6 +38,26 @@ func (c *MockClient) Close(ctx context.Context) error {
 	return nil
 }
 
+// Reset resets MockClient to the initial state
+func (c *MockClient) Reset() {
+	c.lock.Lock()
```

Comment:
won't this block if the lock is held by another, so a call to Reset could result in a deadlock?

Reply:
Still thinking this will block if another context has the client holding the lock.

Reply:
Since we have only one lock - it should not cause any deadlock. As who ever holds it should eventually let it go.